src/zcbenchmarks.cpp

```diff
@@ -31,6 +31,110 @@
 #include "zcash/IncrementalMerkleTree.hpp"
 
 using namespace libzcash;
+void Shutdownpartial()
+{
+    LogPrintf("%s: In progress...\n", __func__);
+    static CCriticalSection cs_Shutdown;
+    TRY_LOCK(cs_Shutdown, lockShutdown);
+    if (!lockShutdown)
+        return;
+
+    /// Note: Shutdown() must be able to handle cases in which AppInit2() failed part of the way,
+    /// for example if the data directory was found to be locked.
+    /// Be sure that anything that writes files or flushes caches only does this if the respective
+    /// module was initialized.
+   //RenameThread("zcash-shutoff");
+ //   mempool.AddTransactionsUpdated(1);
+
+/*     StopHTTPRPC();
+    StopREST();
+    StopRPC();
+    StopHTTPServer();
+ */
+ #ifdef ENABLE_WALLET
+    if (pwalletMain)
+       pwalletMain->Flush(false);
+#endif
+#ifdef ENABLE_MINING
+ #ifdef ENABLE_WALLET
```

Comment:
Ditto (the `ENABLE_MINING` check should stay however, so the top of the two `GenerateBitcoins` should be used).

Reply:
done